pkg/proxy/iptables/proxier.go

```diff
@@ -569,15 +569,26 @@ func buildEndpointInfoList(endPoints []hostPortInfo, endpointIPs []string) []*en
 	return filteredEndpoints
 }
 
-type healthCheckEndpoints struct {
-	name      types.NamespacedName
-	endpoints []hostPortInfo
+// Map of namespaced names (eg, service namespace/name) to whether the service
+// has local endpoints or not.  Only local endpoints need healthchecking.
+type hcUpdateMap map[types.NamespacedName]sets.String
```

Comment:
Comment is not an accurate description of the type. "Map from a namespaced name to either an empty set or a set containing the same namespaced name as a string, because that's the type we're going to need later, because healthcheck.UpdateEndpoints() is a silly API"...